src/elements/InputDataList.vue

```diff
@@ -0,0 +1,361 @@
+      <datalist :id="datalistId">
+        <option v-for="(item, index) in list" :value="item.value">
+          {{ item.label }}
+        </option>
+      </datalist>
+    </div>
+
+    <div role="alert" class="lux-error" v-if="errormessage">{{ errormessage }}</div>
+    <div class="lux-helper" v-if="helper">{{ helper }}</div>
+  </component>
+</template>
+
+<script>
+/**
+ * DataList inputs are a cross between text inputs and select inputs.
+ * Similar to auto-completion, they are used to offer users suggested values that
+ * filter upon typing, while also allowing them to enter free-form text for the value.
+ *
+ */
+export default {
+  name: "InputDataList",
+  status: "ready",
+  release: "1.0.0",
+  type: "Element",
+  computed: {
+    datalistId() {
+      return this.id + "-list"
+    },
+    hasError() {
+      return this.errormessage.length
+    },
+  },
+  props: {
+    /**
+     * Text value of the form input field.
+     */
+    value: {
+      type: String,
+      default: "",
+    },
+    /**
+     * The placeholder value for the form input field.
+     */
+    placeholder: {
+      type: String,
+      default: "",
+    },
+    /**
+     * The label of the form input field.
+     */
+    label: {
+      type: String,
+      default: "",
+    },
+    /**
+     * List of suggested values.
+     */
+    list: {
+      type: Array,
+      default: [],
+      required: true,
+    },
+    /**
+     * Visually hides the label of the form input field.
+     */
+    hideLabel: {
+      type: Boolean,
+      default: false,
+    },
+    /**
+     * The validation message a user should get.
+     */
+    errormessage: {
+      type: String,
+      default: "",
+    },
+    /**
+     * The helper text a user should get.
+     */
+    helper: {
+      type: String,
+      default: "",
+    },
+    /**
+     * The html element name used for the wrapper.
+     * `div, section`
+     */
+    wrapper: {
+      type: String,
+      default: "div",
+      validator: value => {
+        return value.match(/(div|section)/)
+      },
+    },
+    /**
+     * Unique identifier of the form input field.
+     */
+    id: {
+      type: String,
+      default: "",
+      required: true,
+    },
+    /**
+     * The name attribute for the form input field.
+     */
+    name: {
+      type: String,
+      default: "",
+      required: true,
+    },
+    /**
+     * The width of the form input field.
+     * `auto, expand`
+     */
+    width: {
+      type: String,
+      default: "expand",
+      validator: value => {
+        return value.match(/(auto|expand)/)
+      },
+    },
+    /**
+     * Sets the size of the input area `small, medium, large`
+     */
+    size: {
+      type: String,
+      default: "medium",
+      validator: value => {
+        return value.match(/(small|medium|large)/)
+      },
+    },
+    /**
+     * The maximum number of characters that the user can enter in textarea.
+     */
+    maxlength: {
+      type: String,
```

Comment:
Does this make sense as a String?  I would expect a length to be a number, or have a validator making sure it was a number.

Reply:
Good catch. This property actually shouldn't be on this component. I started by duplicating InputText and its residual.  I will remove it from InputDataList and make `maxlength` a Number in InputText.